sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -649,7 +650,40 @@ def monitoring_infos(self):
       for mi in op.monitoring_infos(transform_id).values():
         fixed_mi = self._fix_output_tags_monitoring_info(transform_id, mi)
         all_monitoring_infos_dict[monitoring_infos.to_key(fixed_mi)] = fixed_mi
-    return list(all_monitoring_infos_dict.values())
+
+    infos_list = list(all_monitoring_infos_dict.values())
+
+    def inject_pcollection_into_element_count(monitoring_info):
+      """
+      If provided metric is element count metric:
+      Finds relevant transform output info in current process_bundle_descriptor
+      and adds tag with PCOLLECTION_LABEL and pcollection_id into monitoring
+      info.
+      """
+      if monitoring_info.urn == monitoring_infos.ELEMENT_COUNT_URN:
+        if not monitoring_infos.PTRANSFORM_LABEL in monitoring_info.labels:
+          return
+        ptransform_label = monitoring_info.labels[
+            monitoring_infos.PTRANSFORM_LABEL]
+        if not monitoring_infos.TAG_LABEL in monitoring_info.labels:
+          return
+        tag_label = monitoring_info.labels[monitoring_infos.TAG_LABEL]
+
+        if not ptransform_label in self.process_bundle_descriptor.transforms:
+          return
+        if not tag_label in self.process_bundle_descriptor.transforms[
+            ptransform_label].outputs:
+          return
+
+        pcollection_name = (self.process_bundle_descriptor
+                            .transforms[ptransform_label].outputs[tag_label])
+        monitoring_info.labels[
```

Comment:
Is the old TAG still in here as a label? Can it be removed?

Reply:
Synced with Alex offline.
My assumption was that it's ok to keep original tags.
However it is worth removing them to reduce amount of data transfered and avoid situations when wrong fields are used downstream.